### PR TITLE
Step Scaling Validation

### PR DESCRIPTION
**Pull request by knguyen100000010**
- Added validation for step scaling configurations. Tested.
- Fix a small issue with button hiding (add button should only be displayed when target tracking scaling tab is selected).

### CODE REVIEW

File: deploy-board/deploy_board/templates/groups/asg_policy.tmpl [hunk 1]
```diff
@@ -455,6 +453,159 @@ aria-hidden="true">
...
+
+            function validateAndParseSteps(stepString, isUp, isStep) {
+                if (stepString.length == 0) {
+                    alert("Steps or adjustments cannot be empty");
+                    return null;
+                }
+
+                var stepArray = stepString.split(',').map(Number);
+                for (i = 0; i < stepArray.length; i++) {
+                    if (isNaN(stepArray[i])) {
```
Comment (rwxzhu) [inline deploy-board/deploy_board/templates/groups/asg_policy.tmpl:549]: nit: can this be Number.isNaN instead? I think the latter is preferred for some type conversion pitfalls w/ the global method, [for ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#description).

**Review by rwxzhu — APPROVED** [review]
LGTM, just a couple of nits.

**Review by rwxzhu — APPROVED** [review]

**Comment by ntascii** [comment]
> Added validation for step scaling configurations. Tested.
Can you add a test case and steps you've taken to check your work?

File: deploy-board/deploy_board/templates/groups/asg_policy.tmpl [hunk 2]
```diff
@@ -455,6 +453,159 @@ aria-hidden="true">
         </div>
 
         <script>
+            function validateStepScalingConfig(stepScalingConfig) {
```
Comment (ntascii) [inline deploy-board/deploy_board/templates/groups/asg_policy.tmpl:456]: should these js code be in a separate .js file? the template file is getting lengthy

**Pull request merged**